### PR TITLE
ADEN-9580 Introducing CCPA basic mechanism

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: src/ccpa/UserSignalMechanism.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+});
+
+function createPrivacyString(explicitNotice = USP_VALUES.na, optOutSale = USP_VALUES.na) {
+    return `${USP_VERSION}${explicitNotice}${optOutSale}`;
+}
+
+class UserSignalMechanism {
+    mounted = false;
+
+    static installStub() {
```
Comment (Bielik20) [inline src/ccpa/UserSignalMechanism.js]: I would move queueing functionality to separate file and reuse it across GDPR and CCPA.

File: src/ccpa/UserSignalMechanism.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+            } else if (commandName === 'getQueue') {
+                callback(queue, true);
+            } else {
+                queue.push([commandName, version, callback]);
+            }
+        };
+    }
+
+    constructor(options) {
+        this.options = Object.assign(getDefaultOptions(), options);
```
Comment (Bielik20) [inline src/ccpa/UserSignalMechanism.js]: I assume there is no spread operator support on the platform?

File: src/ccpa/UserSignalMechanism.js [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+        console.log('CCPA: User Signal Mechanism initialized');
+    }
+
+    configure(options) {
+        Object.assign(this.options, options);
+    }
+
+    install() {
+        if (this.mounted) {
```
Comment (Bielik20) [inline src/ccpa/UserSignalMechanism.js]: Is there a scenario when this would happen?

Comment (dmnsobczak) [reply src/ccpa/UserSignalMechanism.js]: Right now: I think no.

File: src/ccpa/UserSignalMechanism.js (hunk 1 above)
Comment (dmnsobczak) [reply src/ccpa/UserSignalMechanism.js]: 👍

**Pull request merged**